src/main/java/com/github/elenterius/biomancy/init/AcidInteractions.java

```diff
@@ -157,4 +165,55 @@ else if (livingEntity.tickCount % 10 == 0 && livingEntity.getRandom().nextFloat(
 		}
 	}
 
+	private static final String TIMER_KEY = "biomancy:digestion_timer";
+	private static final float EFFICIENCY = 0.8f;
+	public static void tryDigest(ItemEntity itemEntity) {
+		if (!digestible(itemEntity)) return;
+		CompoundTag data = itemEntity.getPersistentData();
+		if (data.contains(TIMER_KEY) && data.getInt(TIMER_KEY) < 100) {
+			data.putInt(TIMER_KEY,data.getInt(TIMER_KEY)+1);
+			//TODO: Bubble particles? Couldn't get it working for some reason.
+			// Copied the addParticle call from handleEntityInsideAcid and it said no.
```

Comment:
`addParticle` is only implemented by the client level and this method is called on the server level side

Reply:
It also doesn't help that the bubble particle dissapears next tick when not in a water block 🤦